scripts/tests/test_sh_to_aodf.py

```diff
@@ -8,128 +8,88 @@
+                            '--sh_basis', 'descoteaux07_legacy', '-f',
+                            '--include_center',
                             print_result=True, shell=True)
 
     assert ret.success
-    mock_filtering.assert_called_once()
 
-    ret_sym_fodf = nib.load("out_sym.nii.gz")
-    test_sym_fodf = nib.load(sym_fodf)
-    assert np.allclose(ret_sym_fodf.get_fdata(), test_sym_fodf.get_fdata())
-
-
-@pytest.mark.parametrize("in_fodf,out_fodf",
-    [[os.path.join(data_path, "fodf_descoteaux07_sub_full.nii.gz"),
-      os.path.join(data_path, "fodf_descoteaux07_sub_twice.nii.gz")]])
-def test_asym_input(script_runner, mock_filtering, in_fodf, out_fodf):
-    os.chdir(os.path.expanduser(tmp_dir.name))
-
-    ret = script_runner.run('scil_sh_to_aodf.py',
-                            in_fodf,
-                            'out_fodf3.nii.gz',
-                            '--sphere', 'repulsion100',
-                            '--sigma_angular', '1.0',
-                            '--sigma_spatial', '1.0',
-                            '--sigma_range', '1.0',
-                            '--sh_basis', 'descoteaux07', '-f',
-                            print_result=True, shell=True)
-
-    assert ret.success
-    mock_filtering.assert_called_once()
-    
-    ret_fodf = nib.load("out_fodf3.nii.gz")
-    test_fodf = nib.load(out_fodf)
+    ret_fodf = nib.load("out_fodf1.nii.gz")
+    test_fodf = nib.load(expected_fodf)
     assert np.allclose(ret_fodf.get_fdata(), test_fodf.get_fdata())
 
 
-@pytest.mark.parametrize("in_fodf,out_fodf",
-    [[os.path.join(data_path, 'fodf_descoteaux07_sub.nii.gz'),
-      os.path.join(data_path, 'fodf_descoteaux07_sub_full.nii.gz')]])
-def test_cosine_method(script_runner, mock_filtering, in_fodf, out_fodf):
+@pytest.mark.parametrize("in_fodf,out_fodf", [
+    [os.path.join(test_data_root, 'fodf_descoteaux07_sub.nii.gz'),
+     os.path.join(test_data_root,
+                  'fodf_descoteaux07_sub_cosine_asym.nii.gz')]])
+def test_cosine_method(script_runner, in_fodf, out_fodf):
     os.chdir(os.path.expanduser(tmp_dir.name))
 
     ret = script_runner.run('scil_sh_to_aodf.py',
                             in_fodf, 'out_fodf1.nii.gz',
                             '--sphere', 'repulsion100',
-                            '--method', 'cosine',
-                            '--sh_basis', 'descoteaux07',
-                            '-f',
+                            '--method', 'cosine', '-f',
+                            '--sh_basis', 'descoteaux07_legacy',
                             print_result=True, shell=True)
 
     assert ret.success
 
-    # method cosine is fast and not mocked
-    mock_filtering.assert_not_called()
-
     ret_fodf = nib.load("out_fodf1.nii.gz")
     test_fodf = nib.load(out_fodf)
 
-    # We expect the output to be different from the
-    # one obtained with angle-aware bilateral filtering
-    assert not np.allclose(ret_fodf.get_fdata(), test_fodf.get_fdata())
```

Comment:
Would it be possible to add something like, if opencl is installed, do another test with --use_opencl? This way, we could test it locally, at least.

Reply:
added a test for gpu. Asserts that it fails when opencl is not installed and asserts that the output is alright when opencl is installed.